pkg/kn/commands/service_create.go

```diff
@@ -69,11 +81,17 @@ func NewServiceCreateCommand(p *KnParams) *cobra.Command {
 			if err != nil {
 				return err
 			}
+			// check if --force flag is given
+			if force, err := cmd.Flags().GetBool("force"); err != nil {
+				return err
+			} else if force {
+				client.Services(namespace).Delete(args[0], &v1.DeleteOptions{})
```

Comment:
What happens if the service does not exist ? In this case, no delete should be attempted.

Reply:
We can check that we `Get` call, here I am not error checking for for `Delete` though, since error is not handled, it creates the service.
@rhuss : Do you have some tweak in mind for this operation ?

Reply:
Don't delete the service here at all. Instead, do a GET, if it exists copy the resourceVersion to &service, and issue an update.

Why? This preserves revision history.

Reply:
Using an update instead of a delete/create would also help with the transactional issue I mentioned (if "delete" works but not "create"), because when the "update" fails you still have the situation as before (e.g. the service running).

The only issue I then have is, that this is not really a `create`, as I think people would expect a clean history for a create. That would again a plus for a 'replace' command or 'update --set' or something. 

For my understanding we should implement:

* `kn service create --force` to create a fresh service without history by deleting any old service of this name. No error if the service already exists.
* `kn service create --replace` for replacing a service, keeping the history, but erroring if the service does not exist.

An alternative for the "replace with keeping history" would be `kn service replace` or `kn update --set`.

Sorry for fueling that discussion again, but I really think that 'create afresh by potentially overwriting an existing service' (good for development) and 'replace everything, potentially with default values, but keep the history' (good for getting back to a defined and known state) are looking similar but are subtly different.

Reply:
> kn service create --replace for replacing a service, keeping the history, but erroring if the service does not exist.

@rhuss ; should it really bother for erroring out ? as in, IMO the `create` verb/command shouldn't block on existence of service, though we could put a proper info on stdout if it existed or not.